nats/src/main/kotlin/rys/ajaxpetproject/nats/controller/RestChatController.kt

```diff
@@ -33,105 +42,88 @@ class RestChatController(private val natsConnection: Connection) {
             }.build()
         }.build()
 
-        val response = NatsValidMongoChatParser.deserializeCreateChatResponse(
+        val response = ChatCreateResponse.parseFrom(
             natsConnection.request(
-                "chat.create",
-                NatsValidMongoChatParser.serializeCreateChatRequest(request)
+                ChatSubjectsV1.ChatRequest.CREATE, request.toByteArray()
             ).get().data
         )
 
         when (response.responseCase) {
-            Mongochat.ChatCreateResponse.ResponseCase.SUCCESS -> {
-
+            ChatCreateResponse.ResponseCase.SUCCESS -> {
                 val success = response.success.result
-
                 val newChat = MongoChat(
                     id = ObjectId(success.id),
                     name = success.name,
                     users = success.usersList.map { ObjectId(it) })
-
                 return ResponseEntity(newChat, HttpStatus.CREATED)
             }
-
-            Mongochat.ChatCreateResponse.ResponseCase.FAILURE -> {
+            ChatCreateResponse.ResponseCase.FAILURE -> {
                 return ResponseEntity(response.failure.message.toString(), HttpStatus.BAD_REQUEST)
             }
-
-            Mongochat.ChatCreateResponse.ResponseCase.RESPONSE_NOT_SET -> {
+            ChatCreateResponse.ResponseCase.RESPONSE_NOT_SET -> {
                 return ResponseEntity("Unexpected internal error. See logs for details", HttpStatus.BAD_REQUEST)
             }
         }
     }
 
     @GetMapping("")
     fun findAllChats(): ResponseEntity<Any> {
-        val response = NatsValidMongoChatParser.deserializeFindChatsResponse(
+        val response = ChatFindAllResponse.parseFrom(
             natsConnection.request(
```

Comment:
why your rest controller invokes nats controller?

Why not the REST controller invoke service layer?